### ISSUE TITLE
CharacterTableDisplayDefaults should be mutable

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#269]
This is a follow-up of PR #249 by @ChrisJefferson: there is just one problem showing up in Jenkins tests in the check of manual examples:
```
============================================================
########> Diff in [ "./../../lib/ctbl.gd", 3458, 3509 ]
# Input is:
CharacterTableDisplayDefaults.User:= rec(
      powermap:= "ATLAS", centralizers:= "ATLAS", chars:= false );;
# Expected output:
# But found:
Error, Records Assignment: <rec> must be a mutable record
########
########> Diff in [ "./../../lib/ctbl.gd", 3458, 3509 ]
# Input is:
Display( CharacterTable( "A5" ) );
# Expected output:
A5

   60  4  3  5  5

p      A  A  A  A
p'     A  A  A  A
   1A 2A 3A 5A B*

# But found:
A5

    2  2  2  .  .  .
    3  1  .  1  .  .
    5  1  .  .  1  1

      1a 2a 3a 5a 5b
   2P 1a 1a 3a 5b 5a
   3P 1a 2a 1a 5b 5a
   5P 1a 2a 3a 1a 1a

X.1     1  1  1  1  1
X.2     3 -1  .  A *A
X.3     3 -1  . *A  A
X.4     4  .  1 -1 -1
X.5     5  1 -1  .  .

A = -E(5)-E(5)^4
 = (1-Sqrt(5))/2 = -b5
########
########> Diff in [ "./../../lib/ctbl.gd", 3458, 3509 ]
# Input is:
Unbind( CharacterTableDisplayDefaults.User );
# Expected output:
# But found:
Error, Record Unbind: <rec> must be a mutable record
########
=========OUTPUT END: testmanuals with autoloaded packages=========
```

**Comment by ChrisJefferson** [comment]
Thanks. I have got a fix for this, but I'm just debating whats the best one. Probably ThreadLocal (so each thread can set the displaydefault differently). This does mean each thread get a copy of the '.Global'. A better option would be to redesign this, but it looks like people (like this test) have been going into this record and directly editing it, so we can't really wrap it sensibly.

**Comment by ChrisJefferson** [comment]
Fixed by #270  (hopefully)

**Comment by alex-konovalov** [comment]
I agree with the fix. It's plausible that someone may want to switch between different threads in the multithreaded UI and use different display options in each of them.

**Comment by alex-konovalov** [comment]
Fixed by #270 - closed.
